sdk/eventhub/event-hubs/review/event-hubs.api.md

```diff
@@ -157,108 +128,82 @@ export class EventPosition {
 }
 
 // @public
-export interface GetPartitionPropertiesOptions extends AbortSignalOptions, ParentSpanOptions {
+export interface InitializationContext extends PartitionContext {
+    setStartPosition(startPosition: EventPosition): void;
 }
 
 // @public
-export interface GetPropertiesOptions extends AbortSignalOptions, ParentSpanOptions {
-}
-
-// @public
-export class InMemoryPartitionManager implements PartitionManager {
-    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
-    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
-    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
-}
-
-// @public
-export interface LastEnqueuedEventInfo {
-    enqueuedTime?: Date;
+export interface LastEnqueuedEventProperties {
+    enqueuedOn?: Date;
     offset?: string;
-    retrievalTime?: Date;
+    retrievedOn?: Date;
     sequenceNumber?: number;
 }
 
 export { MessagingError }
 
 // @public
-export interface ParentSpanOptions {
-    parentSpan?: Span | SpanContext;
+export interface OperationOptions {
+    abortSignal?: AbortSignalLike;
+    tracingOptions?: {
+        spanOptions?: SpanOptions;
+    };
 }
 
 // @public
-export interface PartitionCheckpointer {
+export interface PartitionContext {
+    readonly consumerGroup: string;
+    readonly eventHubName: string;
+    readonly fullyQualifiedNamespace: string;
+    readonly lastEnqueuedEventProperties?: LastEnqueuedEventProperties;
+    readonly partitionId: string;
     updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
-    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
-    // @internal (undocumented)
-    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
 }
 
 // @public
-export interface PartitionContext {
-    consumerGroupName: string;
+export interface PartitionOwnership {
+    consumerGroup: string;
+    etag?: string;
     eventHubName: string;
     fullyQualifiedNamespace: string;
-    partitionId: string;
-}
-
-// @public
-export interface PartitionManager {
-    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
-    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
-    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
-}
-
-// @public
-export interface PartitionOwnership extends PartitionContext {
-    eTag?: string;
-    lastModifiedTimeInMS?: number;
-    offset?: number;
+    lastModifiedTimeInMs?: number;
     ownerId: string;
-    ownerLevel: number;
-    sequenceNumber?: number;
+    partitionId: string;
 }
 
 // @public
 export interface PartitionProperties {
     beginningSequenceNumber: number;
     eventHubName: string;
     lastEnqueuedOffset: number;
+    lastEnqueuedOnUtc: Date;
```

Comment:
Turns out there is an `isEmpty` field that is also part of the PartitionProperties that other languages support.

Reply:
Added. It's coming from an 'any' object so I also watched the breakpoint fire off to see if a value was actually getting filled out for the field (it was). It's a bit difficult to test in our automation since the partitions are never guaranteed to be empty. I do test that it's not false.